src/spatial.rs

```diff
@@ -76,6 +76,29 @@ pub fn next_shapes(dims: &[usize]) -> Vec<Vec<usize>> {
     })
 }
 
+pub fn expand_for_over(old_dims: &[usize]) -> Vec<(Vec<usize>, Vec<usize>)> {
+    let all_next_shapes = next_shapes(old_dims);
+    
+    // Filter to only "over" shapes (same number of dimensions, but values increased)
+    // Special case: [2] should not produce [3] for over operations
+    let over_shapes: Vec<Vec<usize>> = all_next_shapes
```

Comment:
This isn't how this should be handled. 

Split next shapes apart to get the up bits and the over bits in separate calls. Add the special case for [2] to the over shape logic. Not here.